packages/grid/_modules_/grid/components/toolbar/GridToolbarExport.tsx

```diff
@@ -7,28 +7,44 @@ import MenuItem from '@material-ui/core/MenuItem';
 import { isHideMenuKey, isTabKey } from '../../utils/keyboardUtils';
 import { GridApiContext } from '../GridApiContext';
 import { GridMenu } from '../menu/GridMenu';
-import { GridExportOption } from '../../models';
+import { GridExportOption, GridExportFormat, GridExportConfiguration } from '../../models';
 
-export const GridToolbarExport = React.forwardRef<HTMLButtonElement, ButtonProps>(
-  function GridToolbarExport(props, ref) {
+type GridToolbarExportProps = ButtonProps & { exportConfiguration?: GridExportConfiguration };
+
+export const GridToolbarExport = React.forwardRef<HTMLButtonElement, GridToolbarExportProps>(
+  function GridToolbarExport({ exportConfiguration, ...buttonProps }, ref) {
     const apiRef = React.useContext(GridApiContext);
     const exportButtonId = useId();
     const exportMenuId = useId();
     const [anchorEl, setAnchorEl] = React.useState(null);
     const ExportIcon = apiRef!.current.components!.ExportIcon!;
 
-    const ExportOptions: Array<GridExportOption> = [
-      {
+    const ExportOptions: Array<GridExportOption> = [];
+
+    if (exportConfiguration) {
+      if (!exportConfiguration.csv?.disabled) {
+        ExportOptions.push({
+          label: apiRef!.current.getLocaleText('toolbarExportCSV'),
+          format: {
+            name: 'csv',
+            options: exportConfiguration.csv,
+          },
```

Comment:
I think a less nested structure would be better.

```suggestion
          format: 'csv',
          options: exportConfiguration.csv,
```

Reply:
Change applied.